### PR TITLE
atlas: remove usage of tegola.Tile

### PR DESCRIPTION
**Pull request by ear7h**
*note:* this pr had a different name because I originally wrote this as part of the changes in 46e9eeb595b69cfcc897844b28186dc4f1f0d1b7

This pr has two major changes to remove the `tegola.Tile` from the encode function. The first is related to clipping and the second to the epsilon value. I've written the explicit changes below bc they can be hard to follow.

This pr also has new tests for `Encode` that test the clipping mechanism. This required a change in the test provider to be able to specify the features it contains.

## clipping
https://github.com/go-spatial/tegola/blob/46e9eeb595b69cfcc897844b28186dc4f1f0d1b7/tile.go#L101-L104
https://github.com/go-spatial/tegola/blob/46e9eeb595b69cfcc897844b28186dc4f1f0d1b7/tile.go#L192-L194
https://github.com/go-spatial/tegola/blob/46e9eeb595b69cfcc897844b28186dc4f1f0d1b7/atlas/map.go#L208-L213
and the new:
https://github.com/go-spatial/tegola/blob/428bd2c28fa0524cf7e2c147df2d391e7d65fcfe/atlas/map.go#L208-L211

## epsilon
Current:
https://github.com/go-spatial/tegola/blob/46e9eeb595b69cfcc897844b28186dc4f1f0d1b7/tile.go#L209-L225

New:
https://github.com/go-spatial/tegola/blob/428bd2c28fa0524cf7e2c147df2d391e7d65fcfe/maths/simplify/epislon.go#L13-L28

### CODE REVIEW

**Comment by ear7h** [comment]
Since it wasn't much, I also removed the last use of `tegola.Tile` from the atlas package and added it to this PR. However, I ran into some inconsistency between `SeedMapTile` and `PurgeMapTile`, I think they should both use a tile or both use a `zxy`.

https://github.com/go-spatial/tegola/blob/d5f77ab6a487f20cd0efc0ecac16579f47b6a62b/atlas/atlas.go#L105

https://github.com/go-spatial/tegola/blob/d5f77ab6a487f20cd0efc0ecac16579f47b6a62b/atlas/atlas.go#L138

I'm leaning towards the tile...

**Comment by ARolek** [comment]
@ear7h let's use the tile for SeedMapTile.

File: provider/test/provider.go [hunk 1]
```diff
@@ -41,17 +44,42 @@ func (tp *TileProvider) Layers() ([]provider.LayerInfo, error) {
...
-
-	debugTileOutline := provider.Feature{
-		ID:       0,
-		Geometry: ext.AsPolygon(),
-		SRID:     srid,
-		Tags: map[string]interface{}{
-			"type": "debug_buffer_outline",
-		},
-	}
+	if tp.Features != nil {
```
[the remark this answers is not in the document]
Comment (ear7h) [reply provider/test/provider.go]: The former, it was a way to ensure backwards compatibility with other tests and simplicity. I'll update the documentation.

**Comment by ARolek** [comment]
@ear7h this looks great. {lease clean up the commit history and then I will rebase it into v0.11.x

**Comment by ear7h** [comment]
I had to rebase, but I was running into merge issues and simply applied the commit 3ff9ac647c417e30f079d616d703bdf0fad9a69e to `v0.11.x`. GH automatically closed this because I accidentally pushed freshly created branch without applying the commit.
